frontend/src/embedding-sdk-bundle/components/public/dashboard/use-common-dashboard-params.tsx

```diff
@@ -127,19 +132,23 @@ const findDashboardById = (
   dashboardId: DashboardId,
   dashboards: Record<DashboardId, StoreDashboard>,
 ): StoreDashboard | null => {
-  if (typeof dashboardId === "number" || isJWT(dashboardId)) {
-    return dashboards[dashboardId] ?? null;
-  }
-
   // Lookup via entity ids.
   // Dashboards are a mapping of numeric id to dashboard.
-  if (typeof dashboardId === "string") {
+  if (isBaseEntityID(dashboardId)) {
     return (
       Object.values(dashboards).find(
         (dashboard) => dashboard.entity_id === dashboardId,
       ) ?? null
     );
   }
 
+  if (isNumericStringOrNumber(dashboardId) || isJWT(dashboardId)) {
+    return dashboards[dashboardId] ?? null;
+  }
+
   return null;
 };
+
+function isNumericStringOrNumber(value: string | number): boolean {
+  return parseNumber(String(value)) !== null;
```

Comment:
`parseNumber` only accepts a string, so wrapping a value into a string always ensures it can be passed to the function.